ambry-store/src/main/java/com.github.ambry.store/IndexSegment.java

```diff
@@ -699,5 +670,42 @@ public boolean getEntriesSince(StoreKey key, FindEntriesCondition findEntriesCon
     }
     return entries.size() > entriesSizeAtStart;
   }
+
+  /**
+   * @return the prefix for the index and bloom file names.
+   */
+  private String generateIndexBloomFilenamePrefix() {
+    String logSegmentName = startOffset.getName();
+    StringBuilder filenamePrefix = new StringBuilder(logSegmentName);
+    if (!logSegmentName.isEmpty()) {
+      filenamePrefix.append(BlobStore.SEPARATOR);
+    }
+    return filenamePrefix.append(startOffset.getOffset()).append(BlobStore.SEPARATOR).toString();
+  }
+
+  /**
+   * Gets the start {@link Offset} in the {@link Log} that the index with file name {@code filename} represents.
+   * @param filename the name of the index file.
+   * @return the start {@link Offset} in the {@link Log} that the index with file name {@code filename} represents.
+   */
+  static Offset getIndexSegmentStartOffset(String filename) {
+    // file name pattern for index is logSegmentName_offset_index.
+    // If the logSegment name is empty, then the file name pattern is offset_index.
+    String logSegmentName;
+    String startOffsetValue;
+    int firstSepIdx = filename.indexOf(BlobStore.SEPARATOR);
+    int lastSepIdx = filename.lastIndexOf(BlobStore.SEPARATOR);
+    if (firstSepIdx == lastSepIdx) {
+      // pattern is offset_index.
+      logSegmentName = "";
+      startOffsetValue = filename.substring(0, firstSepIdx);
+    } else {
+      // pattern is logSegmentName_offset_index.
+      int lastButOneSepIdx = filename.substring(0, lastSepIdx).lastIndexOf(BlobStore.SEPARATOR);
```

Comment:
logSegmentName itself contains "_". An example log segment name is "0_0". So the index segment that starts at offset 0 will be named "0_0_0_index".